src/Aspire.Cli/Utils/CliHostEnvironment.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Aspire.Cli.Utils;
+
+/// <summary>
+/// Provides information about the CLI host environment capabilities.
+/// </summary>
+internal interface ICliHostEnvironment
+{
+    /// <summary>
+    /// Gets whether the host supports interactive input (e.g., prompts, user input).
+    /// </summary>
+    bool SupportsInteractiveInput { get; }
+
+    /// <summary>
+    /// Gets whether the host supports interactive output (e.g., spinners, progress bars).
+    /// </summary>
+    bool SupportsInteractiveOutput { get; }
+
+    /// <summary>
+    /// Gets whether the host supports colors and ANSI codes.
+    /// </summary>
+    bool SupportsAnsi { get; }
```

Comment:
This is amazing! Thank you, I was going to report an issue about this and I was asking on Aspirifriday's about it as well. 🔥